q2_metadata/normalization/_norm_check_rules.py

```diff
@@ -0,0 +1,311 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from q2_metadata.normalization._norm_errors import (
+    RuleError, ExpectedError, OntologyError, RemapError,
+    ValidationError, NormalizationError,
+    BlankError, MissingError, FormatError
+)
+
+
+def check_rule(variable: str, rules: dict, rule: str, rule_value):
+    """For each rule in the current variable's rules set,
```

Comment:
That would make it a huge class with a lot of useless attributes. The arguments passed to this function are not class attributes (they would never be used again). In fact, these are the parsed rules contents that are checked using functions that are so "static" that I think would make no sens to add inside the class.

Maybe I'm wrong but as a separate module, it did make a lot of sense for Rebecca and @ElDeveloper as we reviewed. I'm in favour of keeping these checking functions in this separate module, notably because it would simplify testing.